galasa-ui/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+'use client';
+
+import React, { createContext, useContext, useState, useEffect } from 'react';
+
+export type ThemeType = 'light' | 'dark' | 'system';
+
+interface ThemeContextType {
+  theme: ThemeType;
+  setTheme: (theme: ThemeType) => void;
+}
+
+const ThemeContext = createContext<ThemeContextType>({
+  theme: 'system',
+  setTheme: () => {},
+});
+
+export const ThemeProvider = ({ children }: { children: React.ReactNode }) => {
+  const [theme, setThemeState] = useState<ThemeType>('system');
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const apply = (t: ThemeType) => {
```

Comment:
Minor - `apply` can be made more descriptive, perhaps like `applyThemeToDocument`?